### PR TITLE
[PATCH v2] example: timer_accuracy: simple timer accuracy measurement app

### PR DESCRIPTION
**Pull request by psavol**
Added simple application to measure timer accuracy. Uses time
API to measure accuracy of received timeouts from timer.

Signed-off-by: Petri Savolainen <petri.savolainen@linaro.org>

### CODE REVIEW

File: example/timer/odp_timer_accuracy.c [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+#include <getopt.h>
+
+#include <odp_api.h>
+
+#define START_OFFSET_NS ((uint64_t)(300 * ODP_TIME_MSEC_IN_NS))
+
+typedef struct test_global_t {
+	struct {
+		long int period_ns;
+		long int res_ns;
```
Comment (muvarov) [inline example/timer/odp_timer_accuracy.c]: to be consistent to period values bellow uint64_t and strtoul() might be better here.

File: example/timer/odp_timer_accuracy.c [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+	/* Init ODP before calling anything else */
+	if (odp_init_global(&instance, init_ptr, NULL)) {
+		printf("Global init failed.\n");
+		return -1;
+	}
+
+	/* Init this thread */
+	if (odp_init_local(instance, ODP_THREAD_CONTROL)) {
+		printf("Local init failed.\n");
+		return -1;
```
Comment (muvarov) [inline example/timer/odp_timer_accuracy.c]: goto 455

File: example/timer/Makefile.am [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+odp_timer_accuracy_SOURCES = odp_timer_accuracy.c
 
 odp_timer_simple_SOURCES = odp_timer_simple.c
 
+odp_timer_test_SOURCES = odp_timer_test.c
+
 if test_example
-TESTS = odp_timer_simple
+TESTS  = odp_timer_accuracy
+TESTS += odp_timer_simple
```
Comment (muvarov) [inline example/timer/Makefile.am]:
```
TESTS  = odp_timer_accuracy \ 
                 odp_timer_simple
```

File: example/timer/odp_timer_accuracy.c [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+	uint64_t         first_ns;
+
+} test_global_t;
+
+static void print_usage(void)
+{
+	printf("\n"
+	       "Timer accuracy test application.\n"
+	       "\n"
+	       "OPTIONS:\n"
```
Comment (muvarov) [inline example/timer/odp_timer_accuracy.c]: optional OPTIONS

File: example/timer/Makefile.am (hunk 3 above)
Comment (psavol) [reply example/timer/Makefile.am]: Fixed in v2.

File: example/timer/odp_timer_accuracy.c (hunk 1 above)
Comment (psavol) [reply example/timer/odp_timer_accuracy.c]: Changed to unsigned long long int and strtoull(). uint64_t != ul, e.g. in a 32bit system ul is 32 bits.

File: example/timer/odp_timer_accuracy.c (hunk 4 above)
Comment (psavol) [reply example/timer/odp_timer_accuracy.c]: Options are by default optional, so didn't change. Some of our applications have exceptional options that are mandatory (e.g. interface name).

File: example/timer/odp_timer_accuracy.c (hunk 2 above)
Comment (psavol) [reply example/timer/odp_timer_accuracy.c]: In practice global/local init failure is fatal. I'd not complicate this simple app with code that tries to survive init failure. Print and exit is enough.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**